rpcs3/Emu/System.cpp

```diff
@@ -727,8 +838,175 @@ game_boot_result Emulator::Load(const std::string& title_id, bool add_only, bool
 			games.reset();
 		}
 
+		m_state_inspection_savestate = g_cfg.savestate.state_inspection_mode.get();
+
+		if (m_ar)
```

Comment:
I think you should move most of the ar stuff to a dedicated file...
This is bloating this file beyond comprehension.
There is little to no benefit in reading all this savestate code if you want to debug actual emulation stuff